annotell-openlabel/annotell/openlabel/models/utils/custom_types.py

```diff
@@ -0,0 +1,31 @@
+
+class Timestamp:
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        if isinstance(v, str):
+            try:
+                return int(v)
+            except ValueError:
+                try:
+                    return float(v)
+                except ValueError:
+                    return v
+
+        if isinstance(v, float):
+            return v
+        return v
```

Comment:
Even if we keep this code in some sense, here are some suggestions to make it more concise.

```suggestion
        if not isinstance(v, str):
            return v

        if v.isdigit():
            return int(v)

        return float(v)
```

Idk how the validation works internally for Pydantic with validators, but don't we have to raise an exception if the validation fails somehow?

In my suggestion the only failure-mode is if the cast to float fails, which seems fine to me.